lib/rules/require-computed-property-dependencies.js

```diff
@@ -297,16 +305,35 @@ module.exports = {
         properties: {
           allowDynamicKeys: {
             type: 'boolean',
+            default: true,
+          },
+          requireServiceNames: {
+            type: 'boolean',
+            default: false,
           },
         },
+        additionalProperties: false,
       },
     ],
   },
 
   ERROR_MESSAGE_NON_STRING_VALUE,
 
   create(context) {
+    const serviceNames = [];
+
     return {
+      Property(node) {
+        // Keep track of injected service names in case we want to ignore them (so they won't be required dependencies).
+        // Note that this assumes services are injected before computed properties for simplicity.
```

Comment:
Is this talking about the order that Ember injects the services, or the order that the services/computeds are declared in the code being linted?

Reply:
Order in the code being linted. I will update the comment to clarify.

```
// Note that for simplicity, we're assuming that service injections come earlier in the file than computed properties.
// (i.e. if the line of code for a service injection comes after a computed property, then we won't be able to ignore it).
```

Reply:
Oof. More importantly, you should include that requirement in the option documentation.

What happens if a normal property comes before the services (or after the services but before the computeds) - will it be falsely treated as a service?  What about when there are no services and/or computeds?

Lastly, does this correctly handle the `myService.property` case (which _should_ be required)?

Reply:
The rule will behave correctly in all of the situations you described. And note that I have a test case for the `myService.property` case.

I added a disclaimer in the documentation about the ordering limitation.

Reply:
@BlueRaja I have good news, I rewrote the change to eliminate the ordering limitation, have a look.

Reply:
Nice! I'm not much of a reviewer, but this LGTM!